packages/manager/apps/key-management-service/src/modules/secret-manager/data/hooks/useCreateSecretVersion.ts

```diff
@@ -0,0 +1,39 @@
+import { ApiError } from '@ovh-ux/manager-core-api';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+  createSecretVersion,
+  CreateSecretVersionParams,
+  CreateSecretVersionResponse,
+  secretVersionsQueryKeys,
+} from '../api/secretVersions';
+import { secretQueryKeys } from '../api/secrets';
+
+export const useCreateSecretVersion = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation<
+    CreateSecretVersionResponse,
+    ApiError,
+    CreateSecretVersionParams
+  >({
+    mutationFn: createSecretVersion,
+    onSuccess: (_, variables) => {
+      // Invalidate secrets
+      queryClient.invalidateQueries({
+        queryKey: secretQueryKeys.list(variables.okmsId),
+        exact: true,
+      });
+      // Invalidate secret details
+      queryClient.invalidateQueries({
+        queryKey: secretQueryKeys.detail(variables.okmsId, variables.path),
+      });
+      // Invalidate secret versions
+      queryClient.invalidateQueries({
+        queryKey: secretVersionsQueryKeys.list(
+          variables.okmsId,
+          variables.path,
+        ),
```

Comment:
As `secretQueryKeys.list` is the parent key of all key is not suffisant to invalidates all queries children ?

Reply:
The first invalidation uses the `exact:true` option to avoid invalidating the cache for all secrets, and invalidate only the list.
So we need to invalidate the secret detail just after.